schemas/input/csl-citation.json

```
@@ -70,85 +70,114 @@
+            { "page": [["I-13", "I-18"]] }
+          ],
+          "anyOf": [
+            { "type": "string" },
+            { "type": "number" },
+            {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/locators"
+              }
+            }
+          ]
+        }
+      ]
+    },
     "locators": {
+      "type": "object",
       "properties": {
         "appendix": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "article": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "book": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "canon": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "chapter": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "column": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "elocation": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "equation": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "figure": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "folio": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "issue": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "line": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "note": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "opus": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "page": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "paragraph": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "part": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "rule": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "section": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "sub-verbo": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "supplement": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "table": {
-          "type": "string"
+          "$ref": "#/definitions/locator-variable"
         },
         "timestamp": {
+          "description": "TODO",
           "type": "string"
```

What should a timestamp format be? Do we care?

If this is specific to a point in multimedia, then it seems we should use a structured format; maybe:

```suggestion
          "description: "An hour, minute, second timestamp for video or audio sources.",
          "type": "array",
          "examples": [
            [1, 12, 32]
          ]
```

---

On timestamps, based on a quick search, the big English language styles use `hour:minute` timestamps.

Also, notable: timestamps are also required as part of item metadata on some resources, like a last updated on an online article. EDTF supports times, we're covered there on the input end. But that raises the question again about the datatype; on an item it's part of an EDTF date. What about here?

What about the styling end? And do these conventions differ by language?

---

I commented in the other issue you opened, but I don't think I've ever seen an editing or retrieval _time_ be cited, only a date. For timestamps in `locator` or `dimensions` (running time), I think the `hh:mm:ss` format is pretty universal).